apis/python/tests/test_basic_xarray_io.py

```diff
@@ -0,0 +1,189 @@
+        assert soma_variable.shape == (8,)
+        assert soma_variable.dims == ("xdim",)
+        assert isinstance(soma_variable._data, xr.core.indexing.LazilyIndexedArray)
+
+    def test_getitem_all(self, soma_variable, numpy_variable):
+        expected = numpy_variable[...]
+
+        actual = soma_variable[...]
+        assert actual.shape == (8,)
+        np.testing.assert_equal(actual.data, expected.data)
+
+        actual = soma_variable[:]
+        assert actual.shape == (8,)
+        np.testing.assert_equal(actual.data, expected.data)
+
+    @pytest.mark.parametrize(
+        "key",
+        [
+            (2,),
+            (-1,),
+            (slice(1, 2),),
+            (slice(None, None),),
+            (slice(1, -1),),
+            (slice(0, 4, 2),),
+            ([0, 3],),
+            ([0, -3],),
+        ],
+    )
+    def test_getitem(self, soma_variable, numpy_variable, key):
+        actual = soma_variable[key]
+        expected = numpy_variable[key]
+
+        assert actual.dims == expected.dims
+        assert actual.shape == expected.shape
+
+        np.testing.assert_equal(actual.data, expected.data)
+
+
+class TestSample3DDenseNDArrayWrapper:
+
+    @pytest.fixture(scope="class")
+    def soma_variable(self, sample_3d_dense_array):
+        array_wrapper = soma_xarray.DenseNDArrayWrapper(sample_3d_dense_array)
+        assert isinstance(array_wrapper, xr.backends.BackendArray)
+        return xr.Variable(
+            ("xdim", "ydim", "zdim"), xr.core.indexing.LazilyIndexedArray(array_wrapper)
+        )
+
+    @pytest.fixture(scope="class")
+    def numpy_variable(self):
+        data = np.reshape(np.arange(64, dtype=np.uint8), (8, 2, 4))
+        return xr.Variable(("xdim", "ydim", "zdim"), data)
+
+    def test_variable_basics(self, soma_variable):
+        assert soma_variable.shape == (8, 2, 4)
+        assert soma_variable.dims == ("xdim", "ydim", "zdim")
+        assert isinstance(soma_variable._data, xr.core.indexing.LazilyIndexedArray)
+
+    def test_getitem_all(self, soma_variable, numpy_variable):
+        expected = numpy_variable[...]
+
+        actual = soma_variable[...]
+        assert actual.shape == (8, 2, 4)
+        np.testing.assert_equal(actual.data, expected.data)
+
+        actual = soma_variable[:]
+        assert actual.shape == (8, 2, 4)
+        np.testing.assert_equal(actual.data, expected.data)
+
+        actual = soma_variable[:, :, :]
+        assert actual.shape == (8, 2, 4)
+        np.testing.assert_equal(actual.data, expected.data)
+
+    @pytest.mark.parametrize(
+        "key",
+        [
+            (2, 1, 3),
+            (0, 0, slice(1, 2)),
+            (2, 1, slice(None, None)),
+            (0, slice(None, None), 1),
+            ([1, 3], 1, 1),
+            ([1, 3], 1, [0, 2]),
+            (-1, -1, -1),
+            (0, 1, slice(0, 4, 2)),
+            ([0, 3, 4], [0, 1, 0], [1, 3, 2]),
+            (slice(0, 5, 2), [0, 1, 0], [1, 3, 2]),
```

Comment:
Could be good to check for ellipses support as well:

```suggestion
            (slice(0, 5, 2), [0, 1, 0], [1, 3, 2]),
            (..., slice(0, 3)),
            (slice(1, 2), ...),
```